src/utils.js

```diff
@@ -123,9 +123,13 @@ go.utils = {
         return moment(date, format, true).isValid();
     },
 
-    is_valid_year: function(input) {
+    is_valid_year: function(input, min, max) {  // expecting string parameters
         // check that it is a number and has four digits
-        return input.length === 4 && go.utils.check_valid_number(input);
+        // AND that the number is within the range determined by the min/max parameters
+        return input.length === 4 && min.length === 4 && max.length === 4 &&
```

Comment:
I think we can drop this requirement now that we're using the minYear and maxYear fields.